docs/jdbc.md

```diff
@@ -0,0 +1,447 @@
+
+Replace these parameters in the example based on your database type:
+
+|               | Mysql & MariaDB                          | Postgres                                 |
+|---------------|------------------------------------------|------------------------------------------|
+| <PROTOCOL>    | mysql                                    | postgresql                               |  
+| <IAM_DB_USER> | my-sa                                    | my-sa@my-project.iam                     |
+| <IAM_EMAIL>   | my-sa@my-project.iam.gserviceaccount.com | my-sa@my-project.iam.gserviceaccount.com |
+
+**Note:** a non-empty string value for the `password` property must be set.
+While this property will be ignored when connecting with the Cloud SQL Connector
+using IAM auth, leaving it empty will cause driver-level validations to fail.
+
+```java
+// Set up URL parameters
+Properties connProps=new Properties();
+connProps.setProperty("user","<IAM_DB_USER>");
+connProps.setProperty("sslmode","disable");
+connProps.setProperty("socketFactory","<DRIVER_CLASS>");
+connProps.setProperty("cloudSqlInstance","project:region:instance");
+connProps.setProperty("enableIamAuth","true");
+connProps.setProperty("cloudSqlTargetPrincipal","<IAM_EMAIL>");
+
+// Initialize connection pool
+HikariConfig config=new HikariConfig();
+config.setJdbcUrl("<JDBC_URL>");
+config.setDataSourceProperties(connProps);
+config.setConnectionTimeout(10000); // 10s
+
+HikariDataSource connectionPool=new HikariDataSource(config);
+```
+
+#### Delegated Service Account Impersonation
+
+In addition, the `cloudSqlDelegates` property controls impersonation delegation.
+The value is a comma-separated list of service accounts containing chained
+list of delegates required to grant the final access_token. If set,
+the sequence of identities must have "Service Account Token Creator" capability
+granted to the preceding identity. For example, if set to
+`"serviceAccountB,serviceAccountC"`, the application default credentials must
+have the Token Creator role on serviceAccountB. serviceAccountB must have
+the Token Creator on serviceAccountC. Finally, C must have Token Creator on
+`cloudSqlTargetPrincipal`. If unset, the application default credential
+principal
+must "Service Account Token Creator" capability granted that role on the
+cloudSqlTargetPrincipal service account.
+
+```java
+connProps.setProperty("cloudSqlTargetPrincipal","TARGET_SERVICE_ACCOUNT");
+connProps.setProperty("cloudSqlDelegates","SERVICE_ACCOUNT_1,SERVICE_ACCOUNT_2");
+```
+
+In this example, the environment's application default principal impersonates
+SERVICE_ACCOUNT_1 which impersonates SERVICE_ACCOUNT_2 which then
+impersonates the TARGET_SERVICE_ACCOUNT.
+
+### Connection via Unix Sockets
+
+To connect using a Unix domain socket (such as the one created by the Cloud SQL
+proxy), you can use the `unixSocketPath` property to specify a path to a local
+file instead of connecting directly over TCP.
+
+##### Mysql
+
+```
```

Comment:
Could we make this a Java string and then plus together the items to make it easier to read?

``` java
String jdbcUrl = "jdbc:mysql:///<DATABASE_NAME>" +
     "?unixSocketPath=</PATH/TO/UNIX/SOCKET>" +
     "&cloudSqlInstance<INSTANCE_CONNECTION_NAME>" +
     "&socketFactory=com.google.cloud.sql.mysql.SocketFactory" +
     "&user=<MYSQL_USER_NAME>"+
     "&password=<MYSQL_USER_PASSWORD>"
```

Reply:
Fixed.